src/persistent.py

```diff
@@ -0,0 +1,36 @@
+import contextlib
+import os
+from typing import Optional
+
+
+class PersistentString:
+    """A string wrapper which is persistent across reboots.
+
+    Methods:
+        get(value):
+            Gets the value of the string
+        set(value):
+            Sets the value of the string and writes to disk
+    """
+
+    def __init__(self, filepath: str):
+        self.filepath = filepath
+        self._value: Optional[str] = None
+        # Load string if it exists
+        with contextlib.suppress(FileNotFoundError):
+            with open(self.filepath) as f:
+                self._value = f.read()
+
+    def set(self, value: Optional[str]):
+        self._value = value
+
+        # Either save string, or delete state file if None
+        if self._value is None:
+            with open(self.filepath, "w") as f:
+                f.write(self._value)
+        else:
+            with contextlib.suppress(FileNotFoundError):
```

Comment:
A slightly more readable way to handle this could be to use `if os.exists(...)`.

Reply:
`os.exists(...)` is _technically_ less safe because the file could be removed in between the check and when the read is finished :^)

If you think we should replace it with a try/except bc nobody knows what contextlib is then I can understand that though lol. I didn't do that because it's longer and a bit uglier

Reply:
While I think that case is pretty unlikely, I admire the defensive programming :)

But I do think a try/except would be more readable. Higher readability often comes at the cost of more lines/lower performance. But it's often worth it if those things aren't mission-critical.

Reply:
Sure, changed it